modules/astrodon-cli/commands/run.ts

```diff
@@ -1,67 +1,118 @@
 import { Develop } from "../../astrodon-build/mod.ts";
 import { AppConfig } from "../../astrodon/mod.ts";
-import { resolve } from "https://deno.land/std@0.122.0/path/mod.ts";
 import { Logger } from "../utils.ts";
+import { dirname, fromFileUrl, join, resolve } from "../deps.ts";
 
 export interface RunOptions {
-  config: string;
+  config?: string;
+  allowAll?: boolean;
+  allowEnv?: string[];
+  allowHrtime?: boolean;
+  allowNet?: string[];
+  allowFFI?: string[];
+  allowRead?: string[];
+  allowWrite?: string[];
+  allowRun?: string[];
+  prompt?: boolean;
 }
 
-const runLogger = new Logger("run");
-
-export async function run(options?: RunOptions, file?: string) {
-
-  let projectInfo: AppConfig = {
-    entry: "",
-    dist: "",
-    info: {
-      name: "Astrodon",
-      id: "Astrodon",
-      copyright: "2022",
-      version: "0.0.0",
-      author: "",
-      shortDescription: "",
-      longDescription: "",
-      homepage: "",
-      icon: [],
-      resources: [],
-      permissions: {
-        allow_hrtime: true,
-        prompt: false,
-      },
-      unstable: false,
+// TODO(marc2332): The default config could inherit some env values such as: user -> author, year -> year
+const DEFAULT_CONFIG: AppConfig = {
+  entry: "",
+  dist: "",
+  info: {
+    name: "Astrodon",
+    id: "Astrodon",
+    copyright: "2022",
+    version: "0.0.0",
+    author: "",
+    shortDescription: "",
+    longDescription: "",
+    homepage: "",
+    icon: [],
+    resources: [],
+    permissions: {
+      allow_hrtime: false,
+      prompt: false,
     },
-  };
+    unstable: false,
+  },
+};
 
-  if (file) {
-    projectInfo.entry = file;
-  } else if (options) {
+const runLogger = new Logger("run");
 
-    // Assume it's a local file by default
-    let filePath = `file://${resolve(Deno.cwd(), options.config)}`;
+async function resolveConfiguration(
+  options: RunOptions,
+  file?: string,
+): Promise<AppConfig | null> {
+  // Default Local path config
+  let configFile = `file://${resolve(Deno.cwd(), "astrodon.config.ts")}`;
 
-    if(options.config.startsWith("http")){
-      filePath = options.config;
-    }
+  if (options.config?.startsWith("http")) {
+    // Custom HTTP path config
+    configFile = options.config;
+  } else if (options.config) {
+    // Custom Local path config
+    configFile = `file://${resolve(Deno.cwd(), options.config)}`;
+  } else if (file?.startsWith("http")) {
+    // Default HTTP path config
+    const remoteDirname = dirname(fromFileUrl(import.meta.url));
+    configFile = `https://${join(remoteDirname, "astrodon.config.ts")}`;
+  }
```

Comment:
Also we need to improve the case where file is not remote and search the config relative to it instead of inside cwd.